docs/drafts/Documentation Guidelines.md

```diff
@@ -0,0 +1,38 @@
+*NB: This information is still in draft form, and should not be relied upon to inform usage of or contribution to `cubing.js`.*
+
+## Purpose of this document
+
+This document serves to describe `cubing.js`'s approach to documentation in both practical and theoretical terms. Anyone looking to contribute either documentation or new features to `cubing.js` should consult this document.
+
+## Documentation guidelines:
+
+- Write API reference documentation inline using [TSDoc-formatted](https://tsdoc.org/) comments 
+    - PR's lacking these comments will be considered incomplete
+- PR's do *not* require contributor-generated usage tutorials - API Reference material is sufficient
+- Writing draft documentation is better than writing none at all
+- Documentation can and should begin immediately 
+  - It is *much* easier to write documentation at the time of coding. Additionally, it is much easier to improve imperfect documentation than to write it from scratch.
+
+## Documentation roadmap:
+
+- Devs to create API reference for all new modules/features going forward (prevents growth of documentation backlog)
+- Decide on documentation process and tooling
+- Create library architecture overview, and create API reference for through backlog of undocumented modules/features (serves needs of experienced developers and contributors)
+- Create tutorials/guides for how to apply the library in practice (serves needs of novice developers)
+
+## Style Guidelines
+
+- Technical references (ie, references to files, projects and tools) should be formatted as follows:
+  - Use backticks \` when there is no accompanying link (eg, "`cubing.js` is a project which aims to...")
+  - Omit backticks when there is an accompanying link (eg, "Make sure you have [node](https://nodejs.org/en/) installed before...")
+  - Reasoning: Universal use of backticks disguises links, leading to confusing navigation
```

Comment:
I've used backticks inside link text for a lot of https://github.com/cubing . I'm open to change, but I'd prefer to keep things consistent unless there's a really good reason to change. Do you have a reference to the issues this can cause?

Reply:
I've realised that the problem I experienced doesn't happen in Github itself, only in my VSCode Markdown previewer.

Github indicates backticks with more than just text highlighting, so the same issue doesn't apply. For example, this text shows all 3 formatting combinations clearly:
![image](https://user-images.githubusercontent.com/52967253/132697309-cc3ed78c-ac8c-4ba6-9954-ff3538841574.png)

However, in VSCode, it only uses text highlighting, making the link on `src/sites/alpha.twizzle.net` ambiguous. (Obviously one can infer from context with it being a URL, but even then it could be plaintext and not a hyperlink):
![image](https://user-images.githubusercontent.com/52967253/132697613-fd8f97f7-fc42-492b-a796-1f4620bbdbd3.png)

Based on this, I think we can ignore this suggestion.